lib/galaxy/datatypes/data.py

```diff
@@ -243,7 +243,10 @@ def missing_meta(self, dataset, check=None, skip=None):
                 continue
             if not check and len(skip) == 0 and dataset.metadata.spec[key].get("optional"):
                 continue  # we skip check for optional and nonrequested values here
-            if not dataset.metadata.element_is_set(key):
+            if not dataset.metadata.element_is_set(key) and (check or dataset.metadata.spec[key].check_required_metadata):
```

Comment:
What I meant is to keep the strict check, but add a profile check. So metadata will be checked only for tools with profile >= 22.01. Then wenn do not need the additional variable check_required_metadata and old tools will still run. 

In https://github.com/galaxyproject/galaxy/pull/13296 we can then take our time and document it the way we want to have this from 22.01. 

Or we just go for the check and doc for 22.05...

Clearly adding a profile seems quite an effort to add to the approx 5 places where the function is called. But in my opinion this is the better way forward. 

Rdata always has a version, but the tool uses the wrong datatype. So making it optional seems wrong.

Reply:
But then that breaks runs on existing data.